bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ *
+ *    Copyright 2021 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.storefrontcontext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultCategory;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultProduct;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultSuggestion;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultsStorefrontContext;
+import com.adobe.cq.commerce.core.search.models.SearchAggregation;
+import com.adobe.cq.commerce.core.search.models.SearchResultsSet;
+
+public class SearchResultsStorefrontContextImpl extends AbstractCommerceStorefrontContext implements SearchResultsStorefrontContext {
+
+    private final SearchResultsSet searchResultsSet;
+
+    public SearchResultsStorefrontContextImpl(SearchResultsSet searchResultsSet) {
+        this.searchResultsSet = searchResultsSet;
+    }
+
+    @Override
+    public String getSearchUnitId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getSearchRequestId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public List<SearchResultProduct> getProducts() {
+        return searchResultsSet.getProductListItems().stream()
+            .map(p -> new SearchResultProduct() {
```

Comment:
Make this a private static nested class and use a constructor reference in the `map()`, like `.map(MyClass::new)`

Reply:
https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/searchResults.ts
I am replicating these types. these need to be available in order to be used with other methods of the MSE api

Reply:
This is fine! 

For the implementation you should prefer static nested classes over (anonymous) inner classes to prevent the implicit reference to the outer object being kept in the inner object. (in some cases this may even lead to memory leaks because the outer object will be retained from GC runs). (effective java 3rd item 24)

Once you have a static nested object, prefer method references over lambdas as this is more expressive.